### DISCUSSION TITLE
Feature: Add tree dependency (child) products to main product

### DISCUSSION THREAD

**Discussion by devmansurov** [lunarphp/lunar#45] [Feature Requests]
Hello. I wanted to share ideas. Could you also add such a feature in which the connection between products for the sale of products with additional dependent products will be implemented. For example, there is a cake and the cake may have different decorations. These decorations will be added as a complementary product to the main cake product.

You can also add it as addon to your repository.

If you have any questions about the idea, write I will be glad to answer. Thanks!

**Comment by alecritson** [comment]
Hi @devmansurov 

Thanks for the contribution, I've converted this to a discussion as I feel it feels outside the realms of an issue :) Least here people can get involved so we can build up an idea of the requirements people might have.

**Reply by devmansurov** [reply]
I think this problem can be solved by using `kalnoy/nestedset` composer package in `products` table like this:

```
Schema::create($this->prefix.'products', function (Blueprint $table) {
    $table->id();
    $table->foreignId('product_type_id')->constrained($this->prefix.'product_types');
    $table->string('status')->index();
    $table->json('attribute_data');
    $table->nestedSet(); // This line maybe solution for child products
    $table->string('brand')->nullable()->index();
    $table->timestamps();
    $table->softDeletes();
});
```

**Comment by alecritson** [comment]
Also just an FYI we have a roadmap set up that we will be using, https://portal.productboard.com/getcandy/1-getcandy, so feel free to post this there :)

**Comment by glennjacobs** [comment]
Not sure I'm understanding the requirements 100%, but is this not associated products? See https://docs.getcandy.io/getcandy/associations.html

If not, then I'm guessing it's something similar to Product Add-ons, Product Bundles or Composite Products. Those GetCandy addons are yet to be developed.

**Reply by devmansurov** [reply]
I don’t know, maybe I couldn’t explain. I'll try to explain it again. For example I am a cake maker. At the same time, I am also selling decorations for cakes as an additional one. There is no problem to add a cake to your platform. But how can I add additional products to the added product (in my case cake) additional products (in my case cake decorations of the client's choice) so that they can be sold together in one order.

**Comment by glennjacobs** [comment]
Ah OK.

So the proper solution would be with a "Product Addons" addon for GetCandy. However, that doesn't exist right now.

So the next best solution would be to use product associations. It would mean the decorations would be on their own cart lines, which may not be ideal. But I guess as you're a developer, you can probably solve that with meta-information on the cart lines, in a parent-child type relationship.

**Reply by devmansurov** [reply]
Yes, of course) I still propose the implementation of this idea. I myself would try to integrate some kind of add-on to your repository, but since I am still a beginner laravle programmer, I did not understand how to properly create add-ons to your repository. Therefore, I need some kind of example or documentation for adding custom addons.

**Reply by glennjacobs** [reply]
It is an addon we will be looking to develop. But it won't be until later in the year, as we're currently concentrating on the core system.

**Reply by devmansurov** [reply]
You're good at. Wish you the best in your journey to create the perfect e-commerce system

**Comment by luizsilva-dev** [comment]
@devmansurov Have you found or developed any solutions?
I need to use it for a pizza place.
So in addition to the traditional pizzas, I have the pizza plain and then the user chooses the toppings.

**Reply by glennjacobs** [reply]
Could you not use product associations?
